src/metabase/pulse.clj

```diff
@@ -114,14 +161,15 @@
 
 (defn- execute-dashboard
   "Fetch all the dashcards in a dashboard for a Pulse, and execute non-text cards"
-  [pulse dashboard & {:as _options}]
+  [{pulse-creator-id :creator_id, :as pulse} dashboard & {:as _options}]
   (let [dashboard-id      (u/the-id dashboard)
         dashcards         (db/select DashboardCard :dashboard_id dashboard-id)
         ordered-dashcards (sort dashcard-comparator dashcards)]
-    (for [dashcard ordered-dashcards
-          :let  [content (dashcard->content dashcard pulse dashboard)]
-          :when (some? content)]
-      content)))
+    (mw.session/with-current-user pulse-creator-id
```

Comment:
This change helps reduce duplication, but now it's not obvious that the current-user is the pulse creator when reading the `can-read?` check in `dashcard-link-card->content` for example. Or when the query is executed for the question cards. I think it makes sense that all content is generated with the creator's permissions, so it does make sense to have that enforced here in `execute-dashboard`. But maybe add some comments or documentation whenever the creator's permissions are used, e.g. in `dashcard-link-card->content`